Cython/Compiler/Optimize.py

```diff
@@ -4473,9 +4473,13 @@ def visit_BinopNode(self, node):
                        getattr(operand2, 'unsigned', '')
             longness = "LL"[:max(len(getattr(operand1, 'longness', '')),
                                  len(getattr(operand2, 'longness', '')))]
+            value = hex(int(node.constant_result))
+            if value[-1] in 'lL':
+                # Python 2 fix
+                value = value[:-1]
```

Comment:
This seems worth a helper function. It'd also be easier to remove again in Cython 3.1. A really bad name would be `delpy2`. :)